### DISCUSSION TITLE
Removing engine binaries from Vercel deployments

### DISCUSSION THREAD

**Discussion by kripod** [prisma/prisma#7040] [Q&A] [answered]
I’ve just noticed the 'Caveats when deploying to AWS platforms' deployment guide’s ['Deleting Prisma engine binaries that are not required'](https://www.prisma.io/docs/guides/deployment/deployment-guides/caveats-when-deploying-to-aws-platforms#deleting-prisma-engine-binaries-that-are-not-required) section and wondering if Vercel removes all the mentioned files to save upon the resulting serverless function size.

Currently, Prisma accounts for >15MB of the final size, which [heavily impacts lambda cold start times](https://mikhail.io/serverless/coldstarts/big3/#does-package-size-matter).

<img width="479" alt="Serverless function size of a web app in production, weighing 17.82MB, of which more than 80% is Prisma" src="https://user-images.githubusercontent.com/14854048/117891592-6ea50580-b2b7-11eb-9024-cd08d9f924ee.png">

**Comment by janpio** [comment, marked as answer]
We are doing as much as we can there, and are actively working on improving this further. Prisma will never be a light dependency, but we are trying our best to minimize the impact. We will hopefully soon be able to publish better documentation on how to understand how exactly serverless functions archives (like the ones built by Vercel) are influenced by Prisma.

To counter the Mikhail link above, in our own investigations (unpublished and we do not trust them fully yet), the impact of the package size on AWS Lambda is actually pretty low (as long you have a few big files and not many, many small ones that all need to be unpacked individually).

**Reply by kripod** [reply]
Thank you for the answer. I’m looking forward to see the extended docs about serverless environment caveats.

I hope that the Vercel team could also help making integration with Prisma more seamless, too.

**Reply by janpio** [reply]
Vercel is fortunately working together with us quite closely. I just got some new toys to play with that will help us understand potential problems in that area.
